solarkraft/src/verify.ts

```diff
@@ -185,6 +234,14 @@ export function verify(args: any) {
         'unverified' // only returned if array is empty
     )
 
+    const monitorStatus =
+        verificationStatus === 'ok'
+            ? MonitorAnalysisStatus.NoViolation
+            : MonitorAnalysisStatus.Violation
+
+    // PR comment: call only if Violation?
+    conditionalAlert(monitorStatus, args.txHash, args.alert)
```

Comment:
```suggestion
    conditionalAlert(monitorStatus, args.txHash, args.alert)
```

I mean, you say in the alert contract that it gets called for each analysis: 🤷 

https://github.com/freespek/solarkraft/blob/417ec2e5ca5fe99cf4255bc047e6b24abb7a7bda/ContractExamples/contracts/alert/src/lib.rs#L26

Reply:
I do, and it does the way it's implemented now, but if you look at the text Igor wrote in #94 it only mentions violations.

Reply:
Igor wrote #94 in 2 minutes, those are not meant as precise descriptions.

Just be consistent across the verifier and the contract.

Reply:
We should ask the actual potential users. I can see how only reporting failures would make sense, but also how reporting both would make sense too.

Reply:
> We should ask the actual potential users. I can see how only reporting failures would make sense, but also how reporting both would make sense too.

The current implementation always calls the contract, which then emits different kinds of events, depending on the status. A user may choose to subscribe to any subset of those. The only time the alert contract stores data is when violations occur.